File: src/main/java/app/askresume/global/util/UriUtil.java
package app.askresume.global.util;

import lombok.experimental.UtilityClass;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.net.URI;

@UtilityClass
public class UriUtil {

    public static URI createUri() {
        return ServletUriComponentsBuilder.fromCurrentRequest()
                .build()
                .toUri();
    }

    public static URI createUri(Long id) {
        return ServletUriComponentsBuilder.fromCurrentRequest()
                .path("/{id}")
                .buildAndExpand(id)
                .toUri();
    }
}

File: src/main/kotlin/app/askresume/global/util/UriUtil.kt
package app.askresume.global.util

import org.springframework.web.servlet.support.ServletUriComponentsBuilder
import java.net.URI

object UriUtil {

    fun createUri(): URI {
        return ServletUriComponentsBuilder.fromCurrentRequest()
            .build()
            .toUri()
    }

    fun createUri(id: Long?): URI {
        return ServletUriComponentsBuilder.fromCurrentRequest()
            .path("/{id}")
            .buildAndExpand(id)
            .toUri()
    }
}